core/app/models/spree/shipping_rate.rb

```diff
@@ -29,7 +29,16 @@ def display_price
     alias_attribute :base_price, :cost
 
     def tax_amount
-      @tax_amount ||= tax_rate.calculator.compute_shipping_rate(self)
+      @_tax_amount ||= begin
+        calculator = if tax_rate.calculator.present?
+                       tax_rate.calculator
```

Comment:
1. Don't use `@_` for variable name. If you want to use memoization use `@taxon_amount ||=`
2. Please try to refactor this method to avoid `begin / end` since it's doesn't look good and breaks SRP since once method does too much. 
For instance, what about something like
```ruby
def tax_amount
  @tax_amount ||= calculator.compute_shipping_rate(self)
end

def calculator
  return tax_rate.calculator if tax_rate.calculator.present?

  Spree::Calculator.with_deleted.find_by(
    calculable_id:   tax_rate.id,
    calculable_type: tax_rate.class.name
  )
end
```